eks/core.py

```diff
@@ -627,3 +633,95 @@ def compute_initial_guesses(ensemble_vars):
     # Compute standard deviation of temporal differences
     std_dev_guess = round(np.std(temporal_diffs_list), 5)
     return std_dev_guess
+
+
+def compute_mahalanobis(x, v, n_latent=3, likelihoods=None, likelihood_threshold=0.9):
```

Comment:
let's move this into a new file `eks/stats.py` just to keep it separate from the kalman code